.github/workflows/ci.yml

```diff
@@ -10,7 +10,7 @@ jobs:
   test:
     strategy:
       matrix:
-        go: [ '1.16.x', '1.17.x', '1.18.x' ]
+        go: [ '1.20.x' ]
```

Comment:
Any special reason you are testing the code against 1.20 only ?

Reply:
The reason is the heavy use of generics. 
I could test if it also works with go 1.19

Reply:
Let's move the `spec.go` to its package so the package will work when it is imported into projects using an older version of go.

Reply:
I just saw, that we can still support go 1.18 
![image](https://user-images.githubusercontent.com/9110370/227203948-4aa4680c-9923-402c-84eb-5858c01eeccb.png)